lib/state-controller.js

```diff
@@ -38,140 +38,57 @@ var StateController = {
 
   downloadedStream: null,
 
-  isKiteSupported: function() {
-    return new Promise((resolve, reject) => {
-      setTimeout(() => {
-        os.platform() === 'darwin' && os.release() >= '10.10.0' ?
-          resolve() :
-          reject({ type: 'bad_state', data: this.STATES.UNSUPPORTED });
-      }, 0);
-    });
+  isKiteSupported () {
+    return os.platform() === 'darwin' && os.release() >= '10.10.0'
+      ? Promise.resolve()
+      : Promise.reject({ type: 'bad_state', data: this.STATES.UNSUPPORTED });
   },
 
-  isKiteInstalled: function() {
-    return new Promise((resolve, reject) => {
-      this.isKiteSupported().then(() => {
-        var ls = child_process.spawnSync('ls', [this.KITE_APP_PATH.installed]);
-        ls.stdout.length !== 0 ?
-          resolve() :
-          reject({ type: 'bad_state', data: this.STATES.UNINSTALLED });
-      }, (err) => {
-        reject(err);
-      });
-    });
+  isKiteInstalled () {
+    return this.isKiteSupported().then(() => {
+      if (!fs.existsSync(this.KITE_APP_PATH.installed)) {
+        throw { type: 'bad_state', data: this.STATES.UNINSTALLED }
+      }
+    })
```

Comment:
Ok, there were too much changes to fix the linter to be in that PR so I created another one just for the linter. I'll merge back master once the linter PR will be merged in master.